### PR TITLE
Add test for ArgsManager::GetChainName

### PR DESCRIPTION
**Pull request by ryanofsky**
There was some test coverage previously, but it was limited and didn't test conflicting and negated arguments.

### CODE REVIEW

File: src/test/util_tests.cpp [hunk 1]
```diff
@@ -794,14 +767,112 @@ BOOST_FIXTURE_TEST_CASE(util_SettingsMerge, SettingsMergeTestingSetup)
...
+        if (!out_file) throw std::system_error(errno, std::generic_category(), "fopen failed");
+    }
+
+    ForEachMergeSetup([&](const ActionList& arg_actions, const ActionList& conf_actions) {
+        TestArgsManager parser;
+        LOCK(parser.cs_args);
+        parser.AddArg("-regtest", "regtest", false, OptionsCategory::OPTIONS);
+        parser.AddArg("-testnet", "testnet", false, OptionsCategory::OPTIONS);
+
+        auto arg = [](Action action) { return action == SET_TEST ? "-testnet" : action == NEGATE_TEST ? "-notestnet" : action == SET_REG ? "-regtest" : action == NEGATE_REG ? "-noregtest" : nullptr; };
```
Comment (MarcoFalke) [inline src/test/util_tests.cpp]: style-nit: Could flow into multiple lines?

File: src/test/util_tests.cpp [hunk 2]
```diff
@@ -600,20 +601,22 @@ BOOST_AUTO_TEST_CASE(util_GetChainName)
...
 //   that aren't sensitive to the network.
 //
-struct SettingsMergeTestingSetup : public BasicTestingSetup {
+struct ArgsMergeTestingSetup : public BasicTestingSetup {
     //! Max number of actions to sequence together. Can decrease this when
     //! debugging to make test results easier to understand.
     static constexpr int MAX_ACTIONS = 3;
 
-    enum Action { SET = 0, NEGATE, SECTION_SET, SECTION_NEGATE, END };
+    enum Action { END, SET, NEGATE, SECTION_SET, SECTION_NEGATE };
```
Comment (MarcoFalke) [inline src/test/util_tests.cpp]: In commit 00093ed7aae76d98792cfe9e3706ec3e27be2a13

Shouldn't this be named (or similar)
```suggestion
    enum Action { NONE, SET, NEGATE, SECTION_SET, SECTION_NEGATE };
```

**Review by maflcko — APPROVED** [review]
ACK, just some style questions. Will scroll through the generated file later today.

**Review by ryanofsky — COMMENTED** [review]
Updated ede18127657652aa1b49b4370f53cb1bfb2e0886 -> 0a82350f92e087ba4dab276215679a1044c8bf2e ([pr/testset2.1](https://github.com/ryanofsky/bitcoin/commits/pr/testset2.1) -> [pr/testset2.2](https://github.com/ryanofsky/bitcoin/commits/pr/testset2.2), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/testset2.1..pr/testset2.2)) with Marco's suggestions (thanks!)

**Comment by MarcoFalke** [comment]
utACK 0a82350f92e087ba4dab276215679a1044c8bf2e

Checked that 4b331159dfa599eced9f9d4e5780173367b43c74 only re-sorts the txt file
Checked that 0a82350f92e087ba4dab276215679a1044c8bf2e produces a nice file which looks plausibly correct (only checked the first 5 lines)

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

utACK 0a82350f92e087ba4dab276215679a1044c8bf2e

Checked that 4b331159dfa599eced9f9d4e5780173367b43c74 only re-sorts the txt file
Checked that 0a82350f92e087ba4dab276215679a1044c8bf2e produces a nice file which looks plausibly correct (only checked the first 5 lines)
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUgagAv7BrgDc1Dcsc7PdRJLiKhqhRZ0UAL5FbNROeLV232+YJHopWYdMJYhvPL7
xvQkMgrLyDy/xvDR74+2CusA0pX+YaAICit9QSV4r+Vb6WwAuuXNB+g757JVvqJI
ietTVrW2zK3sk8x7UWCPssJGUQruXnk3gQM2A7SMBgy8Xs4pPSWgv3CdfctGcyLL
ncoW/wEgRRJgMWJml3NFbdTogD525IoCDpPlOAEAIUwlXTeNzKvC6m2erJ7UaNvt
iof1t5gZOFiPR5s/Gd81BRjpjYuxnJ7xFRd0fUUcuVVmdxiosoEe3u0/xb9RCCRf
PD9Ii+L1Ba8/TAYF8xZ/lxmH+8g+cN/SiJ2fcPM+dAi0wEbBXwYo3py5QYjUbjYO
0G0j/yn95HFZOcyrgKDSaCClA2wXDbWVFdxTMGlzfMza5OG2ZKcxmclLelskiqtY
sUBJGksj4RGgnPZHEJ/NfiSzUGOE7U9rUn5BTvrMxJaoSFJelNeRtrYcrdzwt6M/
J8XA0Ror
=TUSH
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `81dc0a6c7efde6beda8e20d7654843fbdeddec4c54e180c02d9c193b1a3225ef  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e89294010881dc0a6c7efde6beda8e20d7654843fbdeddec4c54e180c02d9c193b1a3225eff0109f66b57d985756f7ac39ca54a38ab54008fff0100509feb515c05a9cecea985964e386f608f01097c90f73fedf7f995b056de6c062ddf808f020f25e206628cfbb2328808e544792123a3a70cf1039bafaffc6118379d53ea6e208f1045cd99a1bf008a8aa4d097d8eca2d0083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6dfff0103d924b4b377f9cd77c2336ccb8c3e39308f1045cd99a1bf0089703d0027c501c180083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6dfff010731c36318e99632f212ae1c9fce3311308f1206e60bd51e953e7dbe78522f4e61c63258aefd4b0a042b8455c2ce69b1743ae2f08f1045cd99a1af008b062b7929ee8e1450083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267f010d302123ecd77eef9667cbe0c89685d1808f1208b946dd2bcb44168c9c18a90fe7fb71bdb9391903608387882b0f206fb368a6808f1045cd99a19f0088f1533ead97d13a40083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

**Comment by ryanofsky** [comment]
Updated 0a82350f92e087ba4dab276215679a1044c8bf2e -> f6bb11fd37f8a2c985786b688ea07699ba75780e ([pr/testset2.2](https://github.com/ryanofsky/bitcoin/commits/pr/testset2.2) -> [pr/testset2.3](https://github.com/ryanofsky/bitcoin/commits/pr/testset2.3), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/testset2.2..pr/testset2.3)) adding test for disabled arguments (`-regtest=0`) in addition to negated arguments (`-noregtest`)

**Comment by MarcoFalke** [comment]
I think we should still test for flags on the command line (I'd doubt that anyone types `-regtest=1` over `-regtest`)

**Comment by ryanofsky** [comment]
> I think we should still test for flags on the command line (I'd doubt that anyone types -regtest=1 over -regtest)

I could add more variations to test this, but the reason I didn't is because `InterpretBool()`, interprets both`"1"` and the empty string as `true`, and we already have tests for variations of `InterpretBool` parsing behavior elsewhere.

https://github.com/bitcoin/bitcoin/blob/e79bbb73e08e3f191e97d3b67a2fbb510c5545ff/src/util/system.cpp#L159-L164

I think the new test will be less fragile if it is only testing merging of overlapping and negated settings and not trying to test boolean parsing at the same time.

Another option would be to test `-regtest` _instead_ of `-regtest=1` not in addition to it. I didn't do this because it would make translating enum values into argument and config strings messier, but I could do it if you'd really prefer it.

**Comment by MarcoFalke** [comment]
re-utACK f6bb11fd37f8a2c985786b688ea07699ba75780e

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

re-utACK f6bb11fd37f8a2c985786b688ea07699ba75780e
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUhZeAv/esRJrFMEBsJ5m0DMtJYJEC8wlFodQn4/2qpURnfd84q3IjefgrFAoZ4N
Iz1f5Q6mLN8HJNyHppbnsqgtO5DWFyH/EuXnKV0/G+qRTreltdGla8mxQ3YIveZ0
hmaa/ReAz3/aCfTQtRqtbXId8cg1+ReemaEIfadtZYOvvEAsgThGM6WQBV6VW9Rb
f1P1jIhkoX4c1VaqC5HLC73ZQk8La2kC/yrG9pHRTX056hpWlf1Yn20dyxwlrcoT
84/JnntV7PXFaxxw0gakrKwUKnckGJwmg0oBMr8XeH01vzE/VXBVizQUFXiRZ8uC
twZ4mAGpOdXtOLqUgTPJgLRQxQnFbnw4RwBaCBD+FtLm6PMJOAzoXxM+356Yk19n
tAoUyuTBLZHvzQ4VICbMzS/KoS6r+UM7ObZpcITgyteV846GuvD0kNXZ+ablOyJQ
BPBYf8Jr/C6bZ2zQrulyTQgiiO2APS5b891Z1saYMauwTOAuK2zqImTarKxgz96/
2/7KVy4H
=IS1+
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `9e787e94a78c234ae577714f7e54430e71958b3215367ff36ed61ff3a94b49ad  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401089e787e94a78c234ae577714f7e54430e71958b3215367ff36ed61ff3a94b49adf01066358073b551fe82081ee639294fb6f508fff01007a4ed9dd379d80c51506afd0c7f3dd708f1045cd9c193f008148949231b27553d0083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6dfff010217043cd45da9fbac1bcbe03e5bfcf4708f120a05589ffe69cde8bfcb207dbb64a776dae81d350b89821c3ba1e6d39ccc6495708f1045cd9c193f008691432642d145c2c0083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff010348b60c59fd436dd6467486e616f569508f0105782e66b418f4d8cb10796d05a0b458e08f120b189963ae2a0d200a62f3da9d88bb1077cc308f6cf4f2f7a962ec4afb3c9148a08f1045cd9c194f00864c20987000624030083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6df01090ca9c3602922d4025adcc564331a9ac08f120fbbc27253c1ff6f71e16f942c778755ac51bcf982b638da6f201cbe11d926d4008f1045cd9c193f008996c8c337ebf77400083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

**Pull request merged**